caddyhttp/fastcgi/fastcgi.go

```diff
@@ -91,14 +126,18 @@ func (h Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) (int, error)
 				resp, err = fcgiBackend.Post(env, r.Method, r.Header.Get("Content-Type"), r.Body, contentLength)
 			}
 
-			if resp.Body != nil {
-				defer resp.Body.Close()
-			}
```

Comment:
We don't if we want to UsePersistentFcgiConnections.

See fcgiclient.go, line 456
`		resp.Body = clientCloser{c, httputil.NewChunkedReader(rb)}`
line 408 ff
```
// clientCloser is a io.ReadCloser. It wraps a io.Reader with a Closer
// that closes FCGIClient connection.
type clientCloser struct {
	*FCGIClient
	io.Reader
}

func (f clientCloser) Close() error { return f.rwc.Close() }
```
i.e. closing the "body" actually closes the connection.

I have replaced this by fastcgi.go lin 139
`				defer fcgiBackend.Close()`